example/extended_ml_workflow.py

```diff
@@ -0,0 +1,130 @@
+"""
+An extended workflow of the toy ML workflow as defined in `example.ml_workflow`.
+The extended example can also return the ROC AUC score and allows users to switch
+to a larger UCI ML covertype dataset, which can be found at:
+http://archive.ics.uci.edu/ml/datasets/Covertype
+"""
+
+import pandas as pd
+from sklearn import datasets, metrics
+from example.ml_workflow import basic_flow
+import time
+
+import bionic as bn
+
+# Convert the example ml workflow to a builder we'll use for extending.
+builder = basic_flow.to_builder()
+
+
+# Predict probabilities for the test data.
+@builder
+def prediction_frame(model, test_frame):
```

Comment:
Maybe at least some of these entities can be included in the original ML workflow? I think at least `prediction_frame` and `precision_recall_frame` should, since they're basically just a refactoring of the original `precision_recall_frame`. But maybe we should just put all of these in the tutorial, and have this file just be an expanded CLI interface for the same workflow. (In fact, we could remove the CLI from the original file, since it's not helpful in the tutorial anyway.)